server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -309,17 +312,72 @@ public void onFailure(Exception e) {
         }
     }
 
-    private void failDecommissionedNodes(ClusterState state) {
+    private void drainNodesWithDecommissionedAttribute(DecommissionRequest decommissionRequest) {
```

Comment:
Can we skip draining if `no_delay` (timeout=0) flag is set and avoid a cluster state update altogether?

Reply:
Added no_delay param for skipping skipping the delay.

Reply:
No can we avoid the `DRAINING` cluster state update altogether and go to `INPROGRESS`

Reply:
I have created a task for the State removal. We have kept it for better debugging in case of decommission request failure

https://github.com/opensearch-project/OpenSearch/issues/4866

Reply:
We have removed Draining state when no_Delay is set to true.